testcontainers-dapr/src/test/java/io/dapr/testcontainers/DaprComponentTest.java

```diff
@@ -0,0 +1,123 @@
+
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Set;
+
+public class DaprComponentTest {
+
+  @Test
+  public void componentStateStoreSerializationTest() {
+    DaprContainer dapr = new DaprContainer("daprio/daprd")
+        .withAppName("dapr-app")
+        .withAppPort(8081)
+        .withComponent(new Component(
+            "statestore",
+            "state.in-memory",
+            "v1",
+            Collections.singletonMap("actorStateStore", new QuotedBoolean("true"))))
+        .withAppChannelAddress("host.testcontainers.internal");
+
+    Set<Component> components = dapr.getComponents();
+    Assert.assertEquals(1, components.size());
+
+    Component kvstore = components.iterator().next();
+    Assert.assertEquals(false, kvstore.getMetadata().isEmpty());
+
+    String componentYaml = dapr.componentToYaml(kvstore);
+    String expectedComponentYaml = "metadata:\n" + "  name: statestore\n"
+        + "apiVersion: dapr.io/v1alpha1\n"
+        + "kind: Component\n"
+        + "spec:\n"
+        + "  metadata:\n"
+        + "  - name: actorStateStore\n"
+        + "    value: \"true\"\n"
+        + "  type: state.in-memory\n"
+        + "  version: v1\n";
+
+    Assert.assertEquals(expectedComponentYaml, componentYaml);
+  }
+
+  @Test
+  public void subscriptionSerializationTest() {
+    DaprContainer dapr = new DaprContainer("daprio/daprd")
+        .withAppName("dapr-app")
+        .withAppPort(8081)
+        .withSubscription("my-subscription", "pubsub", "topic", "/events")
+        .withAppChannelAddress("host.testcontainers.internal");
+
+    Set<Subscription> subscriptions = dapr.getSubscriptions();
+    Assert.assertEquals(1, subscriptions.size());
+
+    String subscriptionYaml = dapr.subscriptionToYaml(subscriptions.iterator().next());
+    String expectedSubscriptionYaml = "metadata:\n" + "  name: my-subscription\n"
+        + "apiVersion: dapr.io/v1alpha1\n"
+        + "kind: Subscription\n"
+        + "spec:\n"
+        + "  route: /events\n"
+        + "  pubsubname: pubsub\n"
+        + "  topic: topic\n";
+    Assert.assertEquals(expectedSubscriptionYaml, subscriptionYaml);
+  }
+
+  @Test
+  public void withComponentFromPath() {
+    URL stateStoreYaml = this.getClass().getClassLoader().getResource("components/statestore.yaml");
+    Path path = Paths.get(stateStoreYaml.getPath());
+
+    DaprContainer dapr = new DaprContainer("daprio/daprd")
+        .withAppName("dapr-app")
+        .withAppPort(8081)
+        .withComponent(path)
+        .withAppChannelAddress("host.testcontainers.internal");
+
+    Set<Component> components = dapr.getComponents();
+    Assert.assertEquals(1, components.size());
+    Component kvstore = components.iterator().next();
+    Assert.assertEquals(false, kvstore.getMetadata().isEmpty());
+
+    String componentYaml = dapr.componentToYaml(kvstore);
+    String expectedComponentYaml = "metadata:\n"
```

Comment:
And here

Reply:
Let's leave this for a `good-first-issue`, so we can help someone new to contribute.